src/api/controllers/interactions.ts

```diff
@@ -4,17 +4,13 @@ import {
   DiscordPayload,
   InteractionCommandPayload,
 } from "../../types/mod.ts";
-import { structures } from "../structures/mod.ts";
 
-export async function handleInternalInteractionCreate(data: DiscordPayload) {
+export function handleInternalInteractionCreate(data: DiscordPayload) {
   if (data.t !== "INTERACTION_CREATE") return;
 
   const payload = data.d as InteractionCommandPayload;
   eventHandlers.interactionCreate?.(
-    {
-      ...payload,
-      member: await structures.createMember(payload.member, payload.guild_id),
```

Comment:
Why remove member from event?

Reply:
`member` still exists in the data, but only the type and value is altered.

Reply:
Well yes member still exists but member does not get cached into memory. Also all other events either create the member or get it from cache. So why removing it here?

Reply:
The type of the first parameter of the "interactionCreate" event is `InteractionCommandPayload`, it includes the "member" property, which is of type `MemberCreatePayload`, **not** `Member`; however, the `createMember()` function resolves to the type `Member`, consequently conflict arises, to resolve this conflict, the raw member payload is passed to the "interactionCreate" event.

Reply:
Then interaction create event should return the interaction and the member imo.

Reply:
Currently, the "interactionCreate" event has one parameter *data* which is of type `InteractionCommandPayload`, however, if I were to remove this, the type "Member" and "MemberCreatePayload" would result in errors.

Reply:
What I mean is not remove it but add a second arg for the member.